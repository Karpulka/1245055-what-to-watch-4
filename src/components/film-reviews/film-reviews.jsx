import React from "react";
import PropTypes from "prop-types";
import comments from "../../mocks/comments";
import Review from "../review/review.jsx";

const FilmReviews = (props) => {
  const {filmID} = props;
  const filmComments = comments[filmID];
  const colDelimeterKey = Math.ceil(filmComments.length / 2);

  return <div className="movie-card__reviews movie-card__row">
    <div className="movie-card__reviews-col">
      {filmComments.slice(0, colDelimeterKey).map((review, i) => <Review review={review} key={review.user + i}/>)}
    </div>
    <div className="movie-card__reviews-col">
      {filmComments.slice(colDelimeterKey).map((review, i) => <Review review={review} key={review.user + i}/>)}
    </div>
  </div>;
};

FilmReviews.propTypes = {
  filmID: PropTypes.number.isRequired
};

export default FilmReviews;
