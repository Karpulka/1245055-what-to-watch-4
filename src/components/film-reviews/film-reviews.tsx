import * as React from "react";
import * as PropTypes from "prop-types";
import ReviewItem from "../review-item/review-item.js";
import {connect} from "react-redux";
import {Operation} from "../../reducer/data/data";
import {getComments} from "../../reducer/data/selectors";

class FilmReviews extends React.PureComponent {
  constructor(props) {
    super(props);

    const {setComments, filmID} = props;
    setComments(filmID);
  }

  render() {
    const {comments} = this.props;

    if (comments && comments.length > 0) {
      const colDelimeterKey = comments ? Math.ceil(comments.length / 2) : 0;

      return <div className="movie-card__reviews movie-card__row">
        <div className="movie-card__reviews-col">
          {comments.slice(0, colDelimeterKey).map((review, i) => <ReviewItem review={review} key={review.user + i}/>)}
        </div>
        <div className="movie-card__reviews-col">
          {comments.slice(colDelimeterKey).map((review, i) => <ReviewItem review={review} key={review.user + i}/>)}
        </div>
      </div>;
    }

    return null;
  }
}

FilmReviews.propTypes = {
  filmID: PropTypes.number.isRequired,
  comments: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    user: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    comment: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired
  })),
  setComments: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  comments: getComments(state)
});

const mapDispatchToProps = (dispatch) => ({
  setComments(filmID) {
    dispatch(Operation.loadComments(filmID));
  }
});

export {FilmReviews};
export default connect(mapStateToProps, mapDispatchToProps)(FilmReviews);
