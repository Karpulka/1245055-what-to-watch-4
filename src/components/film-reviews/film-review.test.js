import React from "react";
import renderer from "react-test-renderer";
import {FilmReviews} from "./film-reviews";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space";

const mockStore = configureStore([]);

const comments = [
  {
    id: 0,
    user: `Kate Muir`,
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 1,
    user: `Kate Muir`,
    rating: 7.5,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: `Kate Muir`,
    rating: 9.1,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  }
];

it(`Render FilmReviews`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      comments
    }
  });

  const tree = renderer
    .create(<Provider store={store}><FilmReviews filmID={1} setComments={() => {}}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
