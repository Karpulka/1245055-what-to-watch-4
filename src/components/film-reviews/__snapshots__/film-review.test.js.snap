// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FilmReviews 1`] = `null`;
